packages/atomic/src/components/common/tabs/tab-bar.tsx

```diff
@@ -0,0 +1,186 @@
+import {h, Component, Element, Host, Listen, State} from '@stencil/core';
+import {Button} from '../button';
+import {TabCommonElement} from './tab-common';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'tab-bar',
+  shadow: true,
+  styleUrl: 'tab-bar.pcss',
+})
+export class TabBar {
+  @Element() private host!: HTMLElement;
+
+  @State()
+  private popoverTabs: typeof Button[] = [];
+
+  private get tabsFromSlot(): TabCommonElement[] {
+    const isTab = (tagName: string) => /atomic-.+-tab$/i.test(tagName);
+    return Array.from(this.host.querySelectorAll<TabCommonElement>('*')).filter(
+      (element) => isTab(element.tagName)
+    );
+  }
+
+  private get selectedTab() {
+    return this.tabsFromSlot.find((tab) => tab.active);
+  }
+
+  private get slotContentWidth() {
+    return this.tabsFromSlot.reduce(
+      (total, el) =>
+        total +
+        parseFloat(window.getComputedStyle(el).getPropertyValue('width')),
+      0
+    );
+  }
+
+  private get containerWidth() {
+    return parseFloat(
+      window.getComputedStyle(this.host).getPropertyValue('width')
+    );
+  }
+
+  private get isOverflow() {
+    return this.slotContentWidth > this.containerWidth;
+  }
+
+  private get tabPopover() {
+    return this.host.shadowRoot?.querySelector('tab-popover');
+  }
+
+  private get popoverWidth() {
+    return this.tabPopover ? this.getElementWidth(this.tabPopover) : 0;
+  }
+
+  private get overflowingTabs() {
+    const containerRelativeRightPosition =
+      this.host.getBoundingClientRect().right;
+    const selectedTabRelativeRightPosition =
+      this.selectedTab?.getBoundingClientRect().right;
+
+    return this.tabsFromSlot.filter((element) => {
+      const isBeforeSelectedTab = selectedTabRelativeRightPosition
+        ? selectedTabRelativeRightPosition >
+          element.getBoundingClientRect().right
+        : null;
```

Comment:
Why are you setting `null` here? `isBeforeSelectedTab` has 3 possible values when only 2 are needed. I'd set it to `false` if possible.